app/src/test/java/com/gruppone/stalker/OrganizationTest.java

```diff
@@ -1,17 +1,70 @@
 package com.gruppone.stalker;
 
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
+import java.util.ArrayList;
+import java.util.List;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
 import org.junit.Test;
 
+
 public class OrganizationTest {
 
   @Test
-  public void isInside() {
+  public void contructor() {
+    //Arrange
+    JSONObject jsonOrg = mock(JSONObject.class);
+    int id = 1;
+    String name = "unipd";
+    JSONArray jsonPlaces = mock(JSONArray.class);
+
+    try {
+      when(jsonOrg.getInt("id")).thenReturn(id);
+      when(jsonOrg.getString("name")).thenReturn(name);
+      when(jsonOrg.getJSONArray("places")).thenReturn(jsonPlaces);
+
+      when(jsonPlaces.length()).thenReturn(0);
+    } catch (JSONException e) {
+      throw new RuntimeException(e);
+    }
+
+    //Act
+    Organization sut = new Organization(jsonOrg);
+
+    //Assert
+    assertEquals(id, sut.getId());
+    assertEquals(name, sut.getName());
+    assertEquals(new ArrayList<Place>(), sut.getPlaces());
```

Comment:
io qua farei una arraylist con dentro un'item nella sezione arrange e poi testerei l'uguaglianza sul contenuto invece che sul fatto che entrambe le arraylist siano vuote

alcuni metodi utili:
https://stackoverflow.com/a/1086720/13323696

forse c'è anche un assertCollection(...) in junit

Reply:
Un item non vuole l'apostrofo, perchè si intende come traducibile a "elemento" o "oggetto".

Comunque francamente questo test lo avevo fatto per avere la coverage prima di correggere le annotation, quindi non lo ho fatto propriamente a regola d'arte. D'altro canto, il fatto che quello che riguarda la lista di places non è testato è segnato su sonarcloud.

Reply:
capito. volevo solo farti presente degli assert sulle collezioni